community/kernel/src/main/java/org/neo4j/kernel/api/ExecutingQueryStatus.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.neo4j.storageengine.api.lock.ResourceType;
+import org.neo4j.time.SystemNanoClock;
+
+abstract class ExecutingQueryStatus
+{
+    abstract long waitTimeNanos( SystemNanoClock clock );
+
+    abstract Map<String,Object> toMap( SystemNanoClock clock );
+
+    static final ExecutingQueryStatus RUNNING = new ExecutingQueryStatus()
+    {
+        @Override
+        long waitTimeNanos( SystemNanoClock clock )
+        {
+            return 0;
+        }
+
+        @Override
+        Map<String,Object> toMap( SystemNanoClock clock )
+        {
+            return Collections.singletonMap( "state", "RUNNING" );
```

Comment:
It is called when the status is inspected, so not on the hot path, but yes, it could be made constant.